src/Dublich/Card.elm

```diff
@@ -0,0 +1,138 @@
+module Dublich.Card exposing (..)
+
+import Expect exposing (Expectation)
+import List.Extra
+
+
+
+-- Define the rank of a card
+
+
+type Rank
+    = Two
+    | Three
+    | Four
+    | Five
+    | Six
+    | Seven
+    | Eight
+    | Nine
+    | Ten
+    | Jack
+    | Queen
+    | King
+    | Ace
+
+
+ranks : List Rank
+ranks =
+    [ Two
+    , Three
+    , Four
+    , Five
+    , Six
+    , Seven
+    , Eight
+    , Nine
+    , Ten
+    , Jack
+    , Queen
+    , King
+    , Ace
+    ]
```

Comment:
When we have a list of all the variants in our custom type, sometimes that list can get out-of-sync when we add a new variant.

It would be nice if the compiler could remind us. Especially so if there are many variants.

There are at least two solutions to this, as described in https://sporto.github.io/elm-patterns/basic/type-iterator.html

The first one is using an [elm-review](https://package.elm-lang.org/packages/jfmengels/elm-review/latest/) rule [NoMissingTypeConstructor](https://package.elm-lang.org/packages/Arkham/elm-review-no-missing-type-constructor/latest/) that automatically checks lists of custom type constructors named all* and warns when they get out of sync.

Reply:
The second one is to get the compiler to remind us of a missing variant by building our list with a case statement. Too much boilerplate we'd better avoid, but if interested check on https://sporto.github.io/elm-patterns/basic/type-iterator.html